crypto/src/main/java/org/hyperledger/besu/crypto/SecurityModule.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.crypto;
+
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * Provides a generic interface for classes which wrap/hide a cryptographic private key. This
+ * interface ensures cryptographic functions required by Ethereum are available to the application
+ * at large, without releasing the content of the private key.
+ */
+public interface SecurityModule {
```

Comment:
This name doesn't quite seem right as it's wrapping a private key, it's not a general security module. Not a great name either but perhaps PrivateKeySecurityModule?